cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/mazha/CouchClient.java

```diff
@@ -108,8 +109,9 @@ private ExecuteResult(InputStream stream,
                 case 3:
                     // 3xx redirection
                     throw new CouchException("Unexpected redirection (3xx) code encountered", responseCode);
-                case 4:
+                case 4: case 5:
                     // 4xx errors normally mean we are not authenticated so we shouldn't retry
+                    // 5xx errors are transient server errors so we should retry
```

Comment:
Wouldn't the else block on 120 take care of that? Perhaps you mean that `fatal` should only be set to `true` if the response code isn't 5xx?